internal/librarian/generate.go

```diff
@@ -250,7 +246,9 @@ func (r *generateRunner) runGenerateCommand(ctx context.Context, libraryID, outp
 
 	// Read the library state from the response.
 	if _, err := readLibraryState(
-		libraryStateLoader,
+		func(data []byte, libraryState *config.LibraryState) error {
```

Comment:
to clarify further. Why can't the json.Unmarshal just be inlined to readLibraryState instead of passing a function?

Reply:
I use a func so that I can simulate an error in unmarshal to improve test coverage.

Reply:
ack if it is just for coverage of one line of code I am not sure it justifies the complexity in my mind. Don't worry about this for now, but may refactor this later.